File: programmers/level1/로또의_최고_순위와_최저_순위/Solution.scala
object Solution {
  def solution(lottos: Vector[Int], win_nums: Vector[Int]): Vector[Int] = {
    var aMatch: Int = 0
    var zeros: Int = 0

    for (lotto <- lottos) {
      if (lotto == 0) {
        zeros += 1
      }
    }

    for (win_num <- win_nums) {
      aMatch += lottos.count(lotto => { lotto == win_num })
    }

    val minRank: Int = if (7 - (aMatch + zeros) < 7) 7 - (aMatch + zeros) else 6
    val maxRank: Int = if (7 - aMatch < 7) 7 - aMatch else 6

    return Vector[Int](minRank, maxRank)
  }

  def main(args: Array[String]): Unit = {
    val lottos: Vector[Int] = Vector(44, 1, 0, 0, 31, 25)
    val win_nums: Vector[Int] = Vector(31, 10, 45, 1, 6, 19)

    println(solution(lottos, win_nums))
  }
}

File: programmers/level1/로또의_최고_순위와_최저_순위/Solution.go
package main

import "fmt"

func solution(lottos []int, win_nums []int) []int {
	match, zeros := 0, 0

	for _, e := range lottos {
		if e == 0 {
			zeros++
		}
	}

	for _, e1 := range win_nums {
		for _, e2 := range lottos {
			if e1 == e2 {
				match++
				break
			}
		}
	}

	maxRank, minRank := 0, 0

	if 7-(match+zeros) < 7 {
		maxRank = 7 - (match + zeros)
	} else {
		maxRank = 6
	}

	if 7-match < 7 {
		minRank = 7 - match
	} else {
		minRank = 6
	}

	return []int{maxRank, minRank}
}

func main() {
	lottos := []int{44, 1, 0, 0, 31, 25}
	win_nums := []int{31, 10, 45, 1, 6, 19}

	fmt.Println(solution(lottos, win_nums))
}
